traceevents/src/main/java/com/tomtom/kotlin/traceevents/TraceEvent.kt

```diff
@@ -80,7 +93,7 @@ public data class TraceEvent(
         if (stackTraceHolder != other.stackTraceHolder) return false
         if (eventName != other.eventName) return false
         if (!args.contentDeepEquals(other.args)) return false
-        if (!parameterNames.contentEquals(other.parameterNames)) return false
+        if (parameterNamesProvider != other.parameterNamesProvider) return false
```

Comment:
Given this is a data class it should have equals and hashCode. It can not use the default generated equals as args is and array of any which requires deep equals.

Having this as a data class is useful in trace event consumers.